### PR TITLE
[Bugfix] Preserve MXFP4 Triton weights in sharded state

### LINKED ISSUE #34448: [BUG] mxfp4 + sharded_state load format silently drops expert weights (gpt-oss-20b)
### Repro script

The script below loads `openai/gpt-oss-20b`, exports it with `save_sharded_state.py` (`--load-format sharded_state`), then compares the total size of the original checkpoint's safetensors against the exported ones. Run from the repo root:

```bash
#!/usr/bin/env bash
set -euo pipefail

MODEL_PATH="${1:?usage: $0 /path/to/openai/gpt-oss-20b [out_dir]}"
OUT_DIR="${2:-$(mktemp -d -t mxfp4-repro-XXXXXX)}"

python3 -c "import sglang; print('sglang version:', sglang.__version__)"
SGLANG_ROOT=$(python3 -c "import os, sglang; print(os.path.dirname(os.path.dirname(os.path.dirname(sglang.__file__))))")

python3 "${SGLANG_ROOT}/examples/runtime/engine/save_sharded_state.py" \
  --model-path "${MODEL_PATH}" \
  --trust-remote-code \
  --tensor-parallel-size 2 \
  --mem-fraction-static 0.7 \
  --attention-backend triton \
  --output "${OUT_DIR}"

echo "original checkpoint (safetensors only):"
du -ch "${MODEL_PATH}"/*.safetensors | tail -1
echo "sharded_state export (safetensors only):"
du -ch "${OUT_DIR}"/*.safetensors | tail -1
```

Run:

```
./repro_mxfp4_sharded_state_loss.sh /path/to/openai/gpt-oss-20b
```

### Result

Output (sglang 0.5.17, `lmsysorg/sglang:latest`, TP=2, GH200):

```
sglang version: 0.5.17
original checkpoint (safetensors only):
13G	total
sharded_state export (safetensors only):
4.0G	total
```

The export is missing every expert's packed FP4 weights (`w13_weight` / `w2_weight`); only the scales, biases, and everything else survive. Loading this checkpoint back with `--load-format sharded_state` runs every expert FFN on zeroed weights, with no error or warning.

### Problem source

This looks like a bug in sglang in the mxfp4 + `ShardedStateLoader` path.

**1. On quantization, the expert weights move off the module.** `Mxfp4MoEMethod.process_weights_after_loading` (triton_kernels branch, in `python/sglang/srt/layers/quantization/mxfp4.py`) moves the quantised weights to a helper object and then deletes them from `layer`, which is the `nn.Module`:

```python
self.w13_weight_triton_tensor = w13_weight
self.w2_weight_triton_tensor = w2_weight
del layer.w13_weight
del layer.w2_weight
```

**2. On save, `ShardedStateLoader.save_model` only sees what's still on the module.** It dumps `layer.state_dict()`, which by this point no longer contains `w13_weight` / `w2_weight`. They live on the quant-method helper object instead, which `state_dict()` never looks at.

**3. On load, the same deletion happens again, before the missing-key check can catch it.** `ShardedStateLoader.load_model` does:

```python
model = _initialize_model(...)
for _, module in model.named_modules():
    quant_method.process_weights_after_loading(module)
```

which zero-initializes the model before its missing-key check runs against it. The check is built from a `state_dict()` that's already missing `w13_weight` / `w2_weight`, so it never flags anything. The model loads "successfully" with those weights left at zero, with no error or warning.

### PR DESCRIPTION
**Pull request by jeremyzhang866**
## Summary

- keep Triton-swizzled MXFP4 expert weights and scales registered on the layer
- serialize non-contiguous postprocessed tensors through contiguous CPU snapshots
- add a CPU regression test for runtime-state aliasing and sharded export

## Root cause

The Triton MXFP4 post-load path moved the swizzled `w13` and `w2` weight wrappers onto the quantization method and deleted the registered layer parameters. `ShardedStateLoader.save_model` only exports `model.state_dict()`, so those runtime weights were omitted.

On reload, SGLang postprocesses freshly initialized placeholders before it builds the expected state dict. The missing weights therefore disappeared from both the checkpoint and the expected-key set, and loading silently succeeded with placeholder runtime weights. The swizzled scales held by `PrecisionConfig` had the same stale-storage problem even though the original scale parameters remained registered.

This change rebinds all four Triton runtime tensors (two weights and two scales) to their existing registered `Parameter` objects. The module state and the kernel wrappers then share the same backing storage, so sharded load copies update the tensors actually consumed by the kernels. Parameter identity and loader attributes are preserved.

The Hopper swizzle can produce non-contiguous runtime tensors. Safetensors rejects those tensors, so the exporter materializes contiguous CPU snapshots at the save boundary without changing the runtime layout.

Fixes #34448

## Validation

- `python3 scripts/ci/check_registered_tests.py`
- Black 26.1.0 (`--check`) on the three changed files
- isort 7.0.0 (`--check-only`) on the new test
- Ruff 0.15.1 (`F401`, `F821`, `UP037`) on the three changed files
- `git diff --check`
- `python3 -m py_compile test/registered/unit/model_loader/test_mxfp4_sharded_state.py`
- focused macOS CPU regression run with import-only stubs for unavailable GPU dependencies: `1 test`, `OK`
- the same regression test against `upstream/main@2d76d537e5` fails as expected because `w13_weight` is absent after postprocessing

The focused CPU test is registered in `base-a-test-cpu`. This local run exercised the real MXFP4 postprocessing method and real sharded-state saver, while stubbing only imports unavailable on macOS. A GH200 `gpt-oss-20b` end-to-end logits comparison was not available locally; project CI and a hardware smoke test should provide the final production validation.

### CODE REVIEW

**Comment by jeremyzhang866** [comment]
The focused regression now passes on this PR and fails against upstream/main@2d76d537e5 at the deleted `w13_weight`, as expected. Official lint is green. Could an authorized maintainer please trigger the base CI (including `base-a-test-cpu`) when convenient? A GH200 `gpt-oss-20b` sharded-state smoke test would also be valuable for the final hardware validation.

**Comment by jeremyzhang866** [comment]
Superseded by #34558, which contains the same fix rebased onto the latest upstream main and uses the requested head branch name without the codex/ prefix.
